homeassistant/components/smappee/__init__.py

```diff
@@ -75,8 +75,22 @@ async def async_setup(hass: HomeAssistant, config: dict):
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Set up Smappee from a zeroconf or config entry."""
     if CONF_IP_ADDRESS in entry.data:
-        smappee_api = api.api.SmappeeLocalApi(ip=entry.data[CONF_IP_ADDRESS])
-        smappee = Smappee(api=smappee_api, serialnumber=entry.data[CONF_SERIALNUMBER])
+        if entry.data[CONF_SERIALNUMBER].startswith("50"):
+            # next generation: local mqtt broker
+            smappee_mqtt = mqtt.SmappeeLocalMqtt(
+                serial_number=entry.data[CONF_SERIALNUMBER]
+            )
+            smappee_mqtt.start()
```

Comment:
Can this do I/o?

Reply:
yes, the start method has I/o. Change to await hass aynsc add exectur job?

Reply:
Make a new function that does both of them so you only have to jump to the executor once.

```python
def _start_mqtt():
    smappee_mqtt.start()
    smappee_mqtt.is_config_ready()

```

Reply:
moved to one method